metals/src/main/scala/scala/meta/internal/metals/ClientExperimentalCapabilities.scala

```diff
@@ -1,58 +1,87 @@
 package scala.meta.internal.metals
-import com.google.gson.JsonElement
-import com.google.gson.JsonNull
+import com.google.gson.JsonObject
 import org.eclipse.{lsp4j => l}
 
+/**
+ * While all these values can be set here, they are here only for
+ * compatablity with clients that were setting them this way. From
+ * a client perspective it's preferable and recommended to use
+ * InitializationOptions instead. From a development perspective
+ * don't add something here unless it's truly a more "experimental"
+ * type feature.
+ */
 final case class ClientExperimentalCapabilities(
-    debuggingProvider: Boolean,
-    decorationProvider: Boolean,
-    didFocusProvider: Boolean,
-    doctorProvider: String,
-    executeClientCommandProvider: Boolean,
-    inputBoxProvider: Boolean,
-    openFilesOnRenameProvider: Boolean,
-    quickPickProvider: Boolean,
-    slowTaskProvider: Boolean,
-    statusBarProvider: String,
-    treeViewProvider: Boolean
+    debuggingProvider: Option[Boolean],
+    decorationProvider: Option[Boolean],
+    didFocusProvider: Option[Boolean],
+    doctorProvider: Option[String],
+    executeClientCommandProvider: Option[Boolean],
+    inputBoxProvider: Option[Boolean],
+    openFilesOnRenameProvider: Option[Boolean],
+    quickPickProvider: Option[Boolean],
+    slowTaskProvider: Option[Boolean],
+    statusBarProvider: Option[String],
+    treeViewProvider: Option[Boolean]
 ) {
-  def this() =
-    this(
-      debuggingProvider = false,
-      decorationProvider = false,
-      didFocusProvider = false,
-      doctorProvider = "html",
-      executeClientCommandProvider = false,
-      inputBoxProvider = false,
-      openFilesOnRenameProvider = false,
-      quickPickProvider = false,
-      slowTaskProvider = false,
-      statusBarProvider = "off",
-      treeViewProvider = false
-    )
-  def doctorFormatIsJson: Boolean = doctorProvider == "json"
-  def statusBarIsOn: Boolean = statusBarProvider == "on"
-  def statusBarIsOff: Boolean = statusBarProvider == "off"
-  def statusBarIsShowMessage: Boolean = statusBarProvider == "show-message"
-  def statusBarIsLogMessage: Boolean = statusBarProvider == "log-message"
+  def doctorFormatIsJson: Boolean = doctorProvider.exists(_ == "json")
+  def statusBarIsOn: Boolean = statusBarProvider.exists(_ == "on")
+  def statusBarIsOff: Boolean = statusBarProvider.exists(_ == "off")
```

Comment:
we are doing && in ClientConfiguration for this configuration but || for all others. does that mean it must be set in all 3 places to be off?

Reply:
Ah, good catch. This is left over from how the logic used to be. Basically "off" was the default, so the reason this was `&&` for this was to ensure that none of them were changed before. Now this can actually be set to `||` since `ClientExperimentalCapabilities` and `InitializationOptions` are no longer defaulting to "off", but rather `None`. I'll change this.